lib/SPIRV/LLVMToSPIRVDbgTran.cpp

```diff
@@ -553,7 +553,14 @@ SPIRVEntry *LLVMToSPIRVDbgTran::transDbgCompileUnit(const DICompileUnit *CU) {
   if (isNonSemanticDebugInfo())
     transformToConstant(
         Ops, {SPIRVDebugInfoVersionIdx, DWARFVersionIdx, LanguageIdx});
-  BM->addModuleProcessed(SPIRVDebug::ProducerPrefix + CU->getProducer().str());
+
+  if (BM->getDebugInfoEIS() == SPIRVEIS_NonSemantic_Shader_DebugInfo_200) {
+    Ops.push_back(BM->getString(CU->getProducer().str())->getId());
+  } else if (BM->getDebugInfoEIS() !=
+             SPIRVEIS_NonSemantic_Shader_DebugInfo_100) {
+    BM->addModuleProcessed(SPIRVDebug::ProducerPrefix +
+                           CU->getProducer().str());
+  }
```

Comment:
This can bring unexpected problems for theoretical SPIRVEIS_NonSemantic_Shader_DebugInfo_300 built on top of 100 and not inheriting features from 200. I suggest to check for OpenCL.DebugInfo.100 explicitly instead of BM->getDebugInfoEIS() !=
             SPIRVEIS_NonSemantic_Shader_DebugInfo_100
             
please also leave TODO to remove old W/A

Reply:
We also have "SPIRV.debug" and "OpenCL.std" debug info. Maybe we should check that the debug info is not NonSemantic (version 200 will be checked twice, but it'd help to avoid issues with the next versions of non-semantic spec)

Reply:
>> Maybe we should check that the debug info is not NonSemantic

This should also work, actually
if (NonSemantic) {
if (200) {
// add producer
} // 200
} // NonSemantic else {
 add W/A
 }

Reply:
Exactly what I meant :) will add shortly